core/unix/os.c

```diff
@@ -8612,8 +8653,9 @@ post_system_call(dcontext_t *dcontext)
              * post-syscall natively anyway, there's no need to restore here.
              */
 #    endif
-            HEAP_TYPE_FREE(dcontext, dcontext->sys_param1, clone3_syscall_args_t,
-                           ACCT_OTHER, PROTECTED);
+            uint app_clone_args_size = (uint)dcontext->sys_param2;
+            heap_free(dcontext, (clone3_syscall_args_t *)dcontext->sys_param1,
```

Comment:
@derekbruening Checking as I'm not completely sure: for the CLONE_VM case, it's probably not sufficient to free this memory just in the parent, and we need to do the same in the child too right?

Reply:
Oh that would be required in the `!CLONE_VM` case.

Reply:
Hmm.. let me fix this now.

Reply:
Done.